File: PRACTICAS/PRACTICA_ACTUAL/JAVA/JAVA/Civitas_P1/src/civitas_p1/CasillaSorpresa.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package civitas_p1;

import java.util.ArrayList;

/**
 *
 * @author javierramirezp
 */
public class CasillaSorpresa extends CasillaDescanso{
   private MazoSorpresas mazo;
   private Sorpresa sorpresa;
   
   public CasillaSorpresa(String nombre, MazoSorpresas mazo){
       super(nombre);
       this.mazo = mazo;
       sorpresa = null;
   }
   @Override
     void recibeJugador(int iactual, ArrayList<Jugador> todos){
    
        if(jugadorCorrecto(iactual,todos)){

            sorpresa = mazo.siguiente();
            informe(iactual,todos);
            sorpresa.aplicarAJugador(iactual, todos);

        }
        
    }
    @Override
    public String toString(){   
        return " llamada " + nombre + " con una sorpresa de tipo " +sorpresa.toString();
    }
}

File: PRACTICAS/PRACTICA_ACTUAL/RUBY/pkg/Civitas-0.0.1/lib/casilla_sorpresa.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.
require_relative 'casilla.rb'
require_relative 'mazo_sorpresas.rb'
require_relative 'sorpresa.rb'


module Civitas
  class Casilla_sorpresa < Casilla
    
    def initialize(n, mazo)
      super(n);
      @mazo = mazo
    end
    
    #Override
    def recibe_jugador(iactual,todos)
       if (jugador_correcto(iactual,todos))
        @sorpresa=@mazo.siguiente
        informe(iactual,todos)
        puts @sorpresa
        @sorpresa.aplicar_a_jugador(iactual,todos)
      end
    end
    
      #Override
    def to_s

      puts "Casilla tipo sorpresa, sorpresa: #{@sorpresa}" 
    end
    
  end
end